docs/developers/README.md

```diff
@@ -195,3 +195,34 @@ but an Error when the sidecar exits with an error. This is only apparent when
 using `kubectl` to get the pods of a TaskRun, not when describing the Pod
 using `kubectl describe pod ...` nor when looking at the TaskRun, but can be quite
 confusing.
+
+## How task results are defined and outputted by a task
+
+Tasks can define results by adding a result on the task spec.
+This is an example:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: print-date
+  annotations:
+    description: |
+      A simple task that prints the date to make sure your cluster / Tekton is working properly.
+spec:
+  results:
+    - name: "current-date"
+      description: "The current date"
+  steps:
+    - name: print-date
+      image: bash:latest
+      args:
+        - "-c"
+        - |
+          date > /tekton/results/current-date
+```
+
+The result is added to a file name with the specified result's name into the `/tekton/results` folder. This is then added to the
+task run status.
```

Comment:
nit: can be done later, it would be nice to show how that part of the TaskRun status looks :angel:

Reply:
This is not part of this PR. The next PR once this one is merged will update the status.

Reply:
oh right, then forget what I just said :joy_cat: